core/src/main/java/bio/terra/pearl/core/service/workflow/EnrollmentService.java

```diff
@@ -142,6 +152,47 @@ public HubResponse<Enrollee> enroll(EnvironmentName envName, String studyShortco
         return hubResponse;
     }
 
+    /**
+     * This method uses the preEnrollResponseId to find the preEnroll Survey for the study and also the user's responses.
+     * It then checks if there is a PROXY answer mapping for any of the questions in the preEnroll survey.
+     * If there is one, it will check the user's response to the question in the PreEnrollment Response.
+     * If the user's response is true, it will return true, otherwise it returns false;
+     *
+     * */
+    private boolean isProxyEnrollment(UUID preEnrollResponseId) {
+        PreEnrollmentResponse preEnrollResponse = preEnrollmentResponseDao.find(preEnrollResponseId).get();
+        if (preEnrollResponse == null || !preEnrollResponse.isQualified()) {
+            return false;
+        }
+        UUID surveyId = preEnrollResponse.getSurveyId();
+        Optional<AnswerMapping> answerMappingForProxyOpt = answerMappingDao.findByTargetField(surveyId, AnswerMappingTargetType.PROXY, "isProxy");
+        if (answerMappingForProxyOpt.isEmpty()) {
+            return false;
+        }
+        String questionStableIdL = answerMappingForProxyOpt.get().getQuestionStableId();
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            try {
+                JsonNode rootNode = objectMapper.readTree(preEnrollResponse.getFullData());
+                if (rootNode.isArray()) {
```

Comment:
at least one, probably two, helper methods would make this code clearer.  e.g. "getAnswerByStableId(String fullDataString)"  -- these methods can be put in `SurveyParseUtils`